### PR TITLE
Implemented Resistance Annotation Parsing

### PR DESCRIPTION
**Pull request by joswinter**
## Resistance Annotation Parsing

I added a mock test file in `test\resources\test_annotations\simple_annotations.txt`.
The mapping from annotation to sequenceSegment will be done in collaboration with the bookmarking part of the application in a future pull request. We are now able to parse annotations using the following code:

```
ResistanceAnnotationParser.getResistanceAnnotationParser().parseAnnotations(annotationFile);
```

---

~~Deadline: 23:59 01-06-2015~~
**Deadline: 23:59 02-06-2015**

### CODE REVIEW

**Comment by albert-smit** [comment]
looks good, but the test data is missing a newline at the end of the file

File: src/main/java/nl/tudelft/lifetiles/annotation/model/ResistanceAnnotationParser.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    private ResistanceAnnotationParser() {
+        // noop
+    }
+
+    /**
+     * Factory method to create a resistance annotation parser.
+     *
+     * @return resistance annotation parser.
+     */
+    public static ResistanceAnnotationParser getResistanceAnnotationParser() {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/annotation/model/ResistanceAnnotationParser.java]: Long method name, rename it to something like `newInstance()`.

**Comment by alangerak** [comment]
Looks good, expect for comments listed above

**Comment by rutgervandenberg** [comment]
You create the appearance of using the factory pattern, but you're actually just using a constructor with a non-default name.

You've got two options: 
- Upgrade to an actual factory (which means add an interface for the parser, an implementation for the parser, and an actual factory)
- Or just make parseAnnotations static and remove  getResistanceAnnotationParser.

**Pull request merged**